SimpleServer/log_collector.h

```diff
@@ -60,32 +61,70 @@ class LogCollectorHTTPServer {
   LogCollectorHTTPServer(int http_port,
                          std::ostream& ostream,
                          const std::string& route = "/log",
-                         const std::string& response_text = "OK\n")
-      : http_port_(http_port), ostream_(ostream), route_(route), response_text_(response_text) {
-    HTTP(http_port_).Register(route_, [this](Request r) {
-      LogEntry entry;
-      entry.t = static_cast<uint64_t>(bricks::time::Now());
-      entry.m = r.method;
-      entry.u = r.url.url_without_parameters;
-      entry.q = r.url.AllQueryParameters();
-      entry.b = r.body;
-      entry.f = r.url.fragment;
-      // TODO(dkorolev): HTTP headers come here.
-      // entry.h = { ... }
-      ostream_ << JSON(entry, "log_entry") << std::endl;
-      r(response_text_);
-    });
+                         const std::string& response_text = "OK\n",
+                         const size_t tick_interval_ms = 1000)  // `0` to disable tick events.
+      : http_port_(http_port),
+        ostream_(ostream),
+        route_(route),
+        response_text_(response_text),
+        tick_interval_ms_(tick_interval_ms),
+        send_ticks_(tick_interval_ms > 0),
+        timer_thread_(&LogCollectorHTTPServer::TimerThreadFunction, this) {
+    HTTP(http_port_)
+        .Register(route_, [this](Request r) {
+          LogEntry entry;
+          entry.t = static_cast<uint64_t>(bricks::time::Now());
```

Comment:
I think the whole part between invoking `bricks::time::Now()` and doing `ostream_ <<` should be under a mutex.

Otherwise there's a nonzero chance for time going back in the output, which should not happen.